src/ModelSchema.php

```diff
@@ -0,0 +1,12 @@
+<?php
```

Comment:
I see this is called the exact same way, every time:

```
$schema = new ModelSchema(new self);
```

Which instantiates TWO new objects... that raised a question mark for me... and it makes me wonder... why don't we move this to the Model instead?

Reply:
Inside the `HasRelationshipFields` trait, for example, we make the `getConnectionAndTable()` method public. What if... instead of doing that... we:
- keep it `private`
- add a new method that IS public (if we need it to), that would return the database schema

After all, you expect a model to be able to return its own schema... right? Rather than having to create an object that does that...

So maybe something like:
```php
public static function getDbTableSchema()
{
    [$connection, $table] = self::getConnectionAndTable();
    $this->schema = new TableSchema($connection->getName(), $table);
}
```

In my eyes this would:
- result in a more intuitive way to get the model schema: `$this->model->getDbTableSchema()`;
- eliminate one file;
- keep the database files model-agnostic;

🤷‍♂️ what do you think @pxpm - feasible or not?